src/lib/voteUtils.ts

```diff
@@ -17,12 +17,40 @@ import { Vote } from "@/app/api/votes/vote";
 export type Support = "AGAINST" | "ABSTAIN" | "FOR";
 
 export function parseSupport(
+  support: string | null,
+  proposalType: ProposalType,
+  start_block: string | null
+): Support {
+  /**
+   * @dev If start_block number of vote is less than 114615036, then it's a proposal
+   *      from old approval voting module where 0 = for, 1 = against
+   *      new approval voting module is 0 = against, 1 = abstain, 2 = for
+   *      note that block number is indicative but works
+   */
+
+  if (start_block && Number(start_block) < 114615036) {
+    return parseSupportOldModule(support, proposalType);
+  }
+
+  switch (Number(support)) {
+    case 0:
+      return "AGAINST";
+    case 1:
+      return "FOR";
+
+    default:
+      return "ABSTAIN";
+  }
+}
```

Comment:
@stepandel New approval module adds the possibility to vote for against (not needed in the ui). Support type is old approval proposal is different from new proposals. Do you confirm? Are there any other areas of the code involved ?

Reply:
just make sure we handle `optimistic` proposal type correctly

and instead of hardcoding the 114615036 directly in voteUtils, I'd add a helper function isOldModule inside `contracts.ts`
so that we can use it in other places if we ever need to
and will determine the block number depending on prod dev and OP vs other deployments